ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -42,7 +42,7 @@ fi
 # Generate working availability zones from the region
 mapfile -t AVAILABILITY_ZONES < <(aws --region "${REGION}" ec2 describe-availability-zones | jq -r '.AvailabilityZones[] | select(.State == "available") | .ZoneName' | sort -u)
 # Generate availability zones with OpenShift Installer required instance types
-mapfile -t INSTANCE_ZONES < <(aws --region "${REGION}" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values="${BOOTSTRAP_NODE_TYPE}","${master_type}","${COMPUTE_NODE_TYPE}" | jq -r '.InstanceTypeOfferings[].Location' | sort -u)
+mapfile -t INSTANCE_ZONES < <(aws --region "${REGION}" ec2 describe-instance-type-offerings --location-type availability-zone --filters Name=instance-type,Values="${BOOTSTRAP_NODE_TYPE}","${master_type}","${COMPUTE_NODE_TYPE}" | jq -r '.InstanceTypeOfferings[].Location' | sort | uniq -d)
```

Comment:
`uniq -d` will fail in two cases:

* Only a single available instance type is requested, e.g. `m5.large,m5.large,null` or `m5.large,null,null`.
* Three separate instance types are requested, and there are regions where only two are available.

I think what we want is some pre-processing to build a `Values` argument without dups or `null`, and then `jq` post-processing to only accept zones which have the expected count of entries.

Reply:
:+1:

Reply:
As `${master_type}` is always set to null in this script https://github.com/openshift/release/blob/829bc9f699bc56277173d9d19ddecc9478e4d65c/ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh#L33   

`${BOOTSTRAP_NODE_TYPE}` is defaulted to `m5.large` https://github.com/openshift/release/pull/14350 
The only var we need to check is `${COMPUTE_NODE_TYPE}`

so either 
| ${BOOTSTRAP_NODE_TYPE} | ${master_type} | ${COMPUTE_NODE_TYPE} |
| --- | --- | --- | --- |
| m5.large | null | m5.large | 
| m5.large | null | null | 
| m5.large | null | m4.xlarge (or some other type) |

if ${COMPUTE_NODE_TYPE} is m5.large or a null then the original sort works
if not then the `uniq -d` sort works

Reply:
actually I'm wrong master_type can change.